lib/rules/lines-around-comment.js

```diff
@@ -207,12 +202,7 @@ module.exports = {
          * @returns {boolean} True if the comment is at parent end.
          */
         function isCommentAtParentEnd(node, nodeType) {
-            const ancestors = context.getAncestors();
-            let parent;
-
-            if (ancestors.length) {
-                parent = ancestors.pop();
-            }
+            const parent = sourceCode.getNodeByRangeIndex(node.range[0]);
```

Comment:
I'm not sure about extracting this, since it really only works for comments - if you had an actual node's range it would return itself!